sql/_13_issues/_23_1h/answers/cbrd_24636.answer

```diff
@@ -0,0 +1,195 @@
+                            class: t node[?]
+                            index: idx_desc term[?] (covers) (desc_index forced)
+                            cost:  ? card ?
+                 cost:  ? card ?
+    sort:  ? desc
+    cost:  ? card ?
+Query stmt:
+select /*+ ORDERED USE_DESC_IDX */ t.y from t_a t_a inner join t_c t on t_a.x=t.x and t_a.x= ?:?  where t.x= ?:?  order by ? desc 
+===================================================
+0
+===================================================
+0
+===================================================
+1
+===================================================
+0
+===================================================
+8
+===================================================
+0
+===================================================
+0
+===================================================
+y    z    
+2     1     
+2     2     
+2     3     
+2     4     
+
+Query plan:
+iscan
+    class: t node[?]
+    index: idx_desc term[?] (covers) (multi_range_opt)
+    cost:  ? card ?
+Query stmt:
+select t.y, t.z from t_c t where ((t.x= ?:? ) or (t.x= ?:? )) order by ? desc , ? for orderby_num()<= ?:? 
+===================================================
+0
+===================================================
+0
+===================================================
+1
+===================================================
+0
+===================================================
+2
+===================================================
+0
+===================================================
+0
+===================================================
+Error:-495
+===================================================
+y    
+1     
+
+Query plan:
+nl-join (cross join)
+    outer: sscan
+               class: t_a node[?]
+               sargs: term[?]
+               cost:  ? card ?
+    inner: iscan
+               class: t node[?]
+               index: idx_desc term[?] (covers) (multi_range_opt)
+               cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+select /*+ ORDERED */ t.y from t_a t_a inner join t_b t on t_a.x=t.x and t_a.x= ?:?  where t.x= ?:?  order by ? for orderby_num()<= ?:? 
+===================================================
+y    
+1     
+
+Query plan:
+temp(order by)
+    subplan: nl-join (cross join)
+                 outer: sscan
+                            class: t_a node[?]
+                            sargs: term[?] AND term[?]
+                            cost:  ? card ?
+                 inner: iscan
+                            class: t node[?]
+                            index: idx_desc term[?]
+                            sargs: term[?]
+                            cost:  ? card ?
+                 cost:  ? card ?
+    sort:  ? asc
+    cost:  ? card ?
+Query stmt:
+select /*+ ORDERED NO_MULTI_RANGE_OPT */ t.y from t_a t_a inner join t_b t on t_a.x=t.x and t_a.x= ?:?  and ?=? where t.x= ?:?  order by ? for orderby_num()<= ?:? 
+===================================================
+0
```

Comment:
This should be done in another issue.
Please exclude it here.

Reply:
I removed the contents from the test case.